copper-server-api/src/main/java/com/pogeyan/cmis/api/data/IObjectFlowService.java

```diff
@@ -1,12 +1,28 @@
 package com.pogeyan.cmis.api.data;
 
+import java.util.List;
 import java.util.Map;
 
 import javax.management.modelmbean.InvalidTargetObjectTypeException;
 
+import org.apache.chemistry.opencmis.commons.data.Acl;
+import org.apache.chemistry.opencmis.commons.data.Properties;
+import org.apache.chemistry.opencmis.commons.enums.UnfileObject;
+import org.apache.chemistry.opencmis.commons.server.ObjectInfoHandler;
+import org.apache.chemistry.opencmis.commons.spi.Holder;
+
 public interface IObjectFlowService {
 	public void setObjectFlowStoreSettings(IObjectFlowStoreSetting dbSettings) throws InvalidTargetObjectTypeException;
 
+	public Boolean beforeCreation(String repositoryId, String folderId, Properties properties, List<String> policies,
+			Acl addAces, Acl removeAces, String userName);
+
+	public Boolean beforeUpdate(String repositoryId, Holder<String> objectId, Holder<String> changeToken,
```

Comment:
- Instead of Holder<String> objectId, can we make it simple String objectId?
- Remove changeToken, objectInfos param